File: 0328-odd-even-linked-list/0328-odd-even-linked-list.py
# Definition for singly-linked list.
# class ListNode:
#     def __init__(self, val=0, next=None):
#         self.val = val
#         self.next = next
class Solution:
    def oddEvenList(self, head: Optional[ListNode]) -> Optional[ListNode]:
        if head is None or head.next is None:
            return head
        e = head
        o = head.next
        o_head = o
        while o and o.next:
            e.next = o.next
            e = o.next
            o.next = e.next
            o = e.next
        e.next = o_head
        return head

File: 0328-odd-even-linked-list/0328-odd-even-linked-list.java
/**
 * Definition for singly-linked list.
 * public class ListNode {
 *     int val;
 *     ListNode next;
 *     ListNode() {}
 *     ListNode(int val) { this.val = val; }
 *     ListNode(int val, ListNode next) { this.val = val; this.next = next; }
 * }
 */
class Solution {
    public ListNode oddEvenList(ListNode head) {
        if (head==null || head.next==null){
            return head;
        }
        ListNode e = head;
        ListNode o = head.next;
        ListNode o_head = o;
        while (o != null && o.next!=null){
            e.next = o.next;
            e = o.next;
            o.next =  e.next;
            o = e.next;
        }
        e.next = o_head;
        return head;
    }
}